enumerables.rb

```diff
@@ -0,0 +1,96 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all?(array = nil)
+    if array
+      my_each { |x| return false unless array == x }
+    elsif !block_given?
+      my_each { |x| return false unless x }
+    else
+      my_each { |x| return false unless yield(x) }
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if args
+      my_each { |x| return false unless args == x }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if yield(x) }
+    end
+    false
+  end
+
+  def my_none?(args = nil, &block)
+    !my_any?(args, &block)
+  end
```

Comment:
- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- I see you are getting my_none? from my_any? so once you solve the issues with your my_any? your my_none? will be solved as well